test/e2e/delete_test.go

```diff
@@ -276,6 +270,45 @@ func (tc *testContext) testWindowsNodeDeletion(t *testing.T) {
 	require.NoError(t, err, "could not delete test namespace")
 }
 
+// deployWorkloadsToDrain creates workloads that require special options to drain
+func (tc *testContext) deployWorkloadsToDrain() error {
+	// Deploy a DaemonSet and wait until its pods have been made ready on each Windows node. DaemonSet pods cannot be
+	// drained from a Node.
+	// Doing this ensures that WMCO is able to handle containers being present on the instance when deconfiguring.
+	ds, err := tc.deployNOOPDaemonSet()
+	if err != nil {
+		return fmt.Errorf("error creating DaemonSet %w", err)
+	}
+	defer tc.client.K8s.AppsV1().DaemonSets(ds.GetNamespace()).Delete(context.TODO(), ds.GetName(),
+		meta.DeleteOptions{})
+	err = tc.waitUntilDaemonsetScaled(ds.GetName(), int(gc.numberOfMachineNodes+gc.numberOfBYOHNodes))
+	if err != nil {
+		return fmt.Errorf("error waiting for DaemonSet pods to become ready: %w", err)
+	}
```

Comment:
nit: this function could logically broken up into 2 helpers. Only a nit from me since this is test code

```
func (tc *testContext) deployWorkloadsToDrain() error {
        if err := tc.deployDaemonSetManagedWorkload(); err != nil {
                return err
        }
        return tc.deployEmptyDirVolumeWorkload()
}
```

Reply:
+1